src/run-in-loop/index.ts

```diff
@@ -45,7 +45,10 @@ export interface RunInLoopOptions {
   initialDelayMs?: number;
 }
 
-export const runInLoop = async (fn: () => Promise<void>, options: RunInLoopOptions) => {
+export const runInLoop = async (
+  fn: () => Promise<{ shouldContinueRunning: boolean } | void>,
```

Comment:
Mhh, why not just let the callback return a `boolean`? Although, `shouldContinueRunning` makes things more explicit so it's not necessarilly bad.